src/cli/api/apihelper.go

```diff
@@ -379,3 +379,85 @@ func (helper *APIHelper) GetMetrics(metricName string, startTime, endTime int64,
 	return hasNext, data, nil
 
 }
+
+func (helper *APIHelper) GetHistory(startTime, endTime int64, desc bool, page uint64) (bool, [][]string, error) {
+
+	hasNext := false
+	if page <= 1 {
+		err := helper.CheckHealth()
+		if err != nil {
+			return hasNext, nil, err
+		}
+	}
+
+	baseURL := helper.Endpoint.URL
+	requestURL := fmt.Sprintf("%s%s", baseURL, strings.Replace(HistoryPath, "{appId}", helper.Client.AppId, -1))
+
+	req, err := http.NewRequest("GET", requestURL, nil)
+	req.Header.Add("Authorization", helper.Client.AuthToken)
+	q := req.URL.Query()
+	if startTime > 0 {
+		q.Add("start-time", strconv.FormatInt(startTime, 10))
+	}
+	if endTime > 0 {
+		q.Add("end-time", strconv.FormatInt(endTime, 10))
+	}
+	if desc {
+		q.Add("order", "desc")
+	} else {
+		q.Add("order", "asc")
+	}
+	q.Add("page", strconv.FormatUint(page, 10))
+	req.URL.RawQuery = q.Encode()
+
+	resp, err := helper.DoRequest(req)
+	if err != nil {
+		return hasNext, nil, err
+	}
+	defer resp.Body.Close()
+
+	raw, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		var errorMsg string
+		switch resp.StatusCode {
+		case 401:
+			errorMsg = fmt.Sprintf(ui.Unauthorized, baseURL, helper.Client.CCAPIEndpoint)
+		default:
+			errorMsg, err = parseErrResponse(raw)
+			if err != nil {
+				return hasNext, nil, err
+			}
+		}
+		return hasNext, nil, errors.New(errorMsg)
+	}
+
+	var history cmodels.HistoryResults
+	err = json.Unmarshal(raw, &history)
+	if err != nil {
+		return hasNext, nil, err
+	}
+
+	var data [][]string
+	for _, entry := range history.Histories {
+		scalingType := "Dynamic"
+		if entry.ScalingType == 1 {
+			scalingType = "Scheduled"
+		}
+		status := "Succeed"
+		if entry.Status == 1 {
+			status = "Failed"
+		}
+		data = append(data, []string{scalingType, status,
+			strconv.Itoa(entry.OldInstances), strconv.Itoa(entry.NewInstances),
+			time.Unix(0, entry.Timestamp).Format(time.RFC3339),
+			entry.Reason, entry.Error,
```

Comment:
why not showing "message" field?  it is useful when the scaling is bounded by min/max

Reply:
I didn't notice the message field is used to display the limitation like this.  
Anyway there was an issue with scaling per min/max limitation.  According to the history record below, the "reason" and "message" fields are conflict with each other

{
  "app_id": "e402a8ee-733c-4b34-b78a-927c0ce9fd91",
  "timestamp": 1517816172461043700,
  "scaling_type": 0,
  "status": 0,
  "old_instances": 19,
  "new_instances": 2,
  "reason": "+1 instance(s) because memoryused > 5MB for 60 seconds",
  "message": "limited by max instances 2",
  "error": ""
}